overlord/snapstate/handlers.go

```diff
@@ -310,20 +334,112 @@ func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName string,
 	return ts, err
 }
 
-func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string, userID int, tm timings.Measurer) error {
+// updates a prerequisite, if it's not providing a content interface that a plug expects it to
+func updatePrereqIfOutdated(t *state.Task, snapName string, contentAttrs []string, userID int, flags Flags) (*state.TaskSet, error) {
+	st := t.State()
+
+	if len(contentAttrs) == 0 {
+		return nil, nil
+	}
+
+	// check if the default provider has all expected content tags
+	if ok, err := hasAllContentAttrs(st, snapName, contentAttrs); err != nil {
+		return nil, err
+	} else if ok {
+		return nil, nil
+	}
+
+	if ok, err := shouldSkipToAvoidConflict(t, snapName); err != nil {
+		return nil, err
+	} else if ok {
+		return nil, nil
+	}
+
+	// default provider is missing some content tags (likely outdated) so update it
+	ts, err := Update(st, snapName, nil, userID, flags)
+	if err != nil {
+		if _, ok := err.(*ChangeConflictError); ok {
+			return nil, &state.Retry{After: prerequisitesRetryTimeout}
+		}
+
+		t.Logf("installed prerequisite '%s' is missing content. Failed to update: %s", snapName, err)
+		return nil, nil
```

Comment:
If it's intentional that we don't return an error, we should have a comment explaining why.